commonmark/pom.xml

```diff
@@ -42,6 +43,19 @@
                     </archive>
                 </configuration>
             </plugin>
+            <plugin>
+                <groupId>org.apache.felix</groupId>
+                <artifactId>maven-bundle-plugin</artifactId>
+                <extensions>true</extensions>
+                <configuration>
+                  <instructions>
+                    <Export-Package>
+                      *,
+                      org.commonmark.internal
+                    </Export-Package>
```

Comment:
What about the other artifacts, don't they need something like this as well? They also have `internal` packages that should not be exported.

Reply:
I haven't got to using extensions yet. If those internal packages are not used by other bundles then no need to export them. I think the whiteboard pattern might fit for extensions. Using that pattern extensions can make themselves available to any user. 

[1] https://en.wikipedia.org/wiki/Whiteboard_Pattern

Reply:
They are not used by other bundles, yeah.